arch/posix/CMakeLists.txt

```diff
@@ -39,11 +39,11 @@ else()
     target_link_options(native_simulator INTERFACE "-m64")
     target_compile_options(native_simulator INTERFACE "-m64")
   else ()
-    zephyr_compile_options(-m32)
+    zephyr_compile_options(-m32 -msse -mfpmath=sse)
```

Comment:
It seems like native_posix could support non-x86 hosts? If so, we'd need a check to make sure these options are valid?

Reply:
ARM (aarch64) will go thru the if in line 30.
In theory we could have this whole branch run only in x86, but I expect in practice this branch is only really used for x86. 
-m32/64 are already machine-dependant options. You could argue they will be accepted in a few more platforms than x86, but none I really expect developers would be using.

Reply:
Could you use zephyr_check_compiler_flag to make sure it's supported?

Reply:
Sure, it's evening here now. I'll change it tomorrow.